ffs/scheduler/scheduler.go

```diff
@@ -176,15 +232,36 @@ func (s *Scheduler) Close() error {
 }
 
 // run spins the long-running goroutines that will execute
-// queued storage and retrieval jobs, renewals and repairs.
+// queued storage, retrieval jobs, renewals, repairs, and gc.
 func (s *Scheduler) run() {
 	defer close(s.finished)
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		if s.gc.AutoGCInterval == 0 {
+			return
+		}
+
+		for {
+			select {
+			case <-s.ctx.Done():
+				return
+			case <-time.After(s.gc.AutoGCInterval):
+				if _, err := s.gcStaged(s.ctx, s.gc.StageGracePeriod); err != nil {
+					log.Errorf("automatic gc: %s", err)
+				}
+			}
+		}
+	}()
```

Comment:
New daemon that runs Staged pins GC.
If `AutoGCInterval` is zero, is considered disabled.